### PR TITLE
update to v7 of Sigstore TUF root

### PR DESCRIPTION
**Pull request by bdehamer**
#### Summary
Updates the embedded Sigstore TUF root to v7. Using the compacted-json form for the embedded root (there's no reason to use the pretty-printed version and it saves 1200 bytes!)

Also, removes an extraneous `tuf.refresh` call (`getTargetInfo` will automatically call `refresh` if necessary)

### CODE REVIEW

**Review by feelepxyz — APPROVED** [review]
👍  Side q: should we look at setting up some automation to update this from the root-signing repo to make sure we're getting the right one and have some verification on these PRs to validate it against the previous one in main?

**Comment by feelepxyz** [comment]
> Using the compacted-json form for the embedded root

ooc is there a compact form out there or did you just minify it?

File: store/public-good-instance-root.json [hunk 1]
```diff
@@ -1,156 +1 @@
...
...9a64473e10ce3ad2f3da372bbdd14b65d07bbaf547e7c8bbbe62b","2e61cd0cbf4a8f45809bda9f7f78c0d33ad11842ff94ae340873e2664dc843de"],"threshold":3},"timestamp":{"keyids":["e1863ba02070322ebc626dcecf9d881a3a38c35c3b41a83765b6ad6c37eaec2a"],"threshold":1}},"consistent_snapshot":true},"signatures":[{"keyid":"25a0eb450fd3ee2bd79218c963dce3f1cc6118badf251bf149f0bd07d5cabe99","sig":"3046022100c0610c0055ce5c4a52d054d7322e7b514d55baf44423d63aa4daa077cc60fd1f022100a097f2803f090fb66c42ead915a2c46ebe7db53a32bf18f2188275cc936f8bdd"},{"keyid":"f5312f542c21273d9485a49394386c4575804770667f2ddb59b3bf0669fddd2f","sig":"304502203134f0468810299d5493a867c40630b341296b92e59c29821311d353343bb3a4022100e667ae3d304e7e3da0894c7425f6b9ecd917106841280e5cf6f3496ad5f8f68e"},{"keyid":"7f7513b25429a64473e10ce3ad2f3da372bbdd14b65d07bbaf547e7c8bbbe62b","sig":"3045022037fe5f45426f21eaaf4730d2136f2b1611d6379688f79b9d1e3f61719997135c022100b63b022d7b79d4694b96f416d88aa4d7b1a3bff8a01f4fb51e0f42137c7d2d06"},{"keyid":"2e61cd0cbf4a8f45809bda9f7f78c0d33ad11842ff94ae340873e2664dc843de","sig":"3044022007cc8fcc4940809f2751ad5b535f4c5f53f5b4952f5b5696b09668e743306ac1022006dfcdf94e94c92163eeb1b47796db62cedaa730aa13aa61b573fe23714730f2"}]}
```
Comment (feelepxyz) [inline store/public-good-instance-root.json:1]: for audit, confirmed it matches: https://tuf-repo-cdn.sigstore.dev/7.root.json

**Comment by bdehamer** [comment]
>  should we look at setting up some automation to update this from the root-signing repo
@feelepxyz yeah, that's a great idea -- it felt a bit weird to just download this file and check it in. I think we could automate the whole thing with a job that runs on some schedule using the existing root to check for a newer one and then open a PR w/ the change.

**Pull request merged**